src/Workspaces/Core/Portable/Shared/Extensions/SemanticModelExtensions.cs

```diff
@@ -80,7 +80,7 @@ public static ISymbol GetEnclosingNamedTypeOrAssembly(this SemanticModel semanti
 
             if (typeInfo.Type != null)
             {
-                return typeInfo.GetTypeWithFlowNullability();
+                return typeInfo.GetTypeWithFlowNullability()!;
```

Comment:
💡 This might be better as:

```csharp
if (typeInfo.GetTypeWithFlowNullability() is { } typeWithFlowNullability)
{
  return typeWithFlowNullability;
}
```

Reply:
`is { }` is never better. :smile:

Reply:
It would also be fine to split it up:

```csharp
var typeWithFlowNullability = typeInfo.GetTypeWithFlowNullability();
if (typeWithFlowNullability is object)
{
  return typeWithFlowNullability;
}
```

Reply:
So GetTypeWithFlowNullability(), once the compiler API is further updated, will be deleted and replaced with just accessing the .Type property, so the code will then get replaced with the obvious `if (typeInfo.Type != null) { return typeInfo.Type }` which the compiler will then properly null-handle. So this will get better, and leaving it in this form eases the conversion.